### PR TITLE
ci: Add Continuous Integration Flag to Run Debug Build

### LINKED ISSUE #10: Have build-artifact.yml workflow also build AddonExample
Often AddonExample gets out of date and stops building because it's not part of the `build.bat` / `build.ps1` build script.  Even if it's not shipped with the release, it'd be nice to build on PRs to make sure it's still compiling.

### PR DESCRIPTION
**Pull request by panicstevenson**
Add ci parameter to the Powershell deployment script, allow the batch
deployment script to pass arguments to Powershell, and add the -ci flag
argument to the build artifact workflow.

Closes #10

### CODE REVIEW

File: AddonExample/AddonExample.csproj [hunk 1]
```diff
@@ -106,7 +106,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json">
-      <Version>12.0.3</Version>
+      <Version>13.0.1</Version>
```
Comment (panicstevenson) [inline AddonExample/AddonExample.csproj:109]: This change can be moved to another pull request if it makes sense. This is to solve the following error:

```
"C:\OverlayPlugin\OverlayPlugin.sln" (default target) (1) ->
"C:\OverlayPlugin\AddonExample\AddonExample.csproj" (default target) (6) ->
(CoreCompile target) ->
  CSC : error CS1705: Assembly 'OverlayPlugin.Common' with identity 'OverlayPlugin.Common, Version=0.18.2.0, Culture=neutral, PublicKeyToken=null' uses 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutra
l, PublicKeyToken=30ad4fe6b2a6aeed' which has a higher version than referenced assembly 'Newtonsoft.Json' with identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6ae
ed' [C:\OverlayPlugin\AddonExample\AddonExample.csproj]
  CSC : error CS1705: Assembly 'OverlayPlugin.Core' with identity 'OverlayPlugin.Core, Version=0.18.2.0, Culture=neutral, PublicKeyToken=null' uses 'Newtonsoft.Json, Version=13.0.0.0, Culture=neutral, P
ublicKeyToken=30ad4fe6b2a6aeed' which has a higher version than referenced assembly 'Newtonsoft.Json' with identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
[C:\OverlayPlugin\AddonExample\AddonExample.csproj]
```

**Comment by quisquous** [comment]
Why debug build? Is debug just the way to build AddonExample? Or do you think it's worth building both?

**Comment by panicstevenson** [comment]
> Why debug build? Is debug just the way to build AddonExample? Or do you think it's worth building both?

The second, Debug is just a way to build AddonExample. It might make sense to just always build both if it's going to block merges? However, It doesn't make sense to add AddonExample to Release in my opinion since the release steps in the Powershell script create the archives from Release/ and I don't think we want to be touching that. Mostly here just looking to avoid breaking any pattern that was previously existing, but maybe that's not the right thing to do.

**Comment by quisquous** [comment]
Yeah, mostly my concern is if building debug is slower?  But this is probably not so large that it matters.

**Review by quisquous — APPROVED** [review]

**Comment by panicstevenson** [comment]
> Yeah, mostly my concern is if building debug is slower? But this is probably not so large that it matters.

It seems to make a very small difference; adding all of the msbuild steps it says it took 77 seconds, and I have a previous run clocked at 70 seconds. We can reassess in the future though if there's optimizations to be made.
